### PR TITLE
Move machine's `output` to top level

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: packages/core/test/types.test.ts [hunk 1]
```diff
@@ -282,19 +258,13 @@ describe('output', () => {
         };
       },
       context: { password: 'okoń' },
```
Comment (davidkpiano) [inline packages/core/test/types.test.ts:260]: What kind of password is this?

**Review by davidkpiano — APPROVED** [review]

**Comment by davidkpiano** [comment]
Does this break existing behavior of `output` on top-level final state appearing in `state.output`?

File: packages/core/test/types.test.ts (hunk 1 above)
Comment (Andarist) [reply packages/core/test/types.test.ts:260]: 😂 I forgot that i sneaked in this polish easter egg here in the past

**Comment by Andarist** [comment]
> Does this break existing behavior of output on top-level final state appearing in state.output?

Yes - I don't see how to maintain the backward compatibility in a sensible way here. This doesn't feel right:
```ts
createMachine({
  types: {} as { output: number },
  initial: 'foo',
  states: {
    foo: {
      type: 'final',
      output: 'bazinga'   
   }
  }
})
```

Configuring some backwards compatible logic at type level (like if `types.output` is not configured) to match what we could figure out here for the runtime world would be quite clunky

**Pull request merged**